pkg/security/controller/namespace_scc_allocation_controller.go

```diff
@@ -98,7 +101,18 @@ func NewNamespaceSCCAllocationController(namespaceInformer corev1informers.Names
 		10*time.Minute,
 	)
 
-	return factory.New().ResyncSchedule(resyncPeriod).WithBareInformers(namespaceInformer.Informer()).WithSyncContext(syncContext).WithSync(c.sync).
+	return factory.New().
+		ResyncSchedule(resyncPeriod).
+		WithBareInformers(namespaceInformer.Informer()).
+		WithSyncContext(syncContext).
+		WithSync(c.sync).
+		WithPostStartHooks(func(ctx context.Context, syncContext factory.SyncContext) error {
```

Comment:
Any particular reason for using post-start hook instead of directly registering metrics in https://github.com/openshift/cluster-policy-controller/blob/master/pkg/cmd/cluster-policy-controller/policy_controller.go#L18 similarly to how we do that for example in https://github.com/openshift/cluster-kube-apiserver-operator/blob/master/pkg/operator/starter.go#L358 ?

Reply:
I have chosen a different approach now